vignettes/sensors.Rmd

```diff
@@ -16,140 +16,179 @@ knitr::opts_chunk$set(
 
 ```{r, warning=FALSE, error=FALSE, prompt=FALSE}
 accel_features <- accelerometer_features(
   accelerometer_data,
   window_length = 128,
   window_overlap = 0.5)
-# Consider a window length of 128 samples
+# Consider a window length of 128 samples,
+# 50% overlap between consecutive windows
 ```
 
 ```{r}
-dplyr::glimpse(accel_features)
+dplyr::glimpse(accel_features$extracted_features)
 ```
+Notice how we now have three extra columns in the feature extraction output namely, window, window_start_time and window_stop_time - which indicate the number of the window in the signal (window 1 : the first window, (0 - window_length) samples), and the start and stop time respectively of that window.
 
-There are many other possible use cases to calculate the accelerometer features. The `time_range` parameter decides the time window to be operated on in the accelerometer signal. The `overlap` parameter decides the amount of overlap between consecutive windows. The `frequency_range` parameter specifies the frequency range to keep for the analysis.
+There are many other possible use cases to calculate the accelerometer features. The `time_filter` parameter decides the time window to be operated on in the accelerometer signal. The `window_overlap` parameter decides the amount of overlap between consecutive windows. The `frequency_filter` parameter specifies the frequency range to keep for the analysis.
 
-```{r, warning=FALSE, error=FALSE, prompt=FALSE}
-accel_features <- accelerometer_features(accelerometer_data, time_filter = c(2,5))
+```{r, warning=FALSE, error=FALSE, prompt=FALSE, eval=FALSE}
+accel_features <- accelerometer_features(accelerometer_data, 
+                                         time_filter = c(2,5))
 # Consider the time ranges between 2 and 5s only
 ```
 
-```{r, warning=FALSE, error=FALSE, prompt=FALSE}
-accel_features <- accelerometer_features(accelerometer_data, frequency_filter = c(4,16))
+```{r, warning=FALSE, error=FALSE, prompt=FALSE, eval=FALSE}
+accel_features <- accelerometer_features(accelerometer_data, 
+                                         frequency_filter = c(4,16))
 # Consider the frequencies only from 4Hz to 16Hz
 ```
 
+There are also advanced processing techniques that you can apply to the signal like detrending (using loess), or decompose the signal into Intrinsic Mode Functions (IMF) using the Hilbert Huang Transform. `detrend` and `IMF` respectively are the parameters that refer to these techniques.
+
+```{r, warning=FALSE, error=FALSE, prompt=FALSE, eval=FALSE}
+accel_features <- accelerometer_features(accelerometer_data, 
+                                         IMF = 2)
+# Consider decomposing the signal into 2 IMFs and then running both the IMFs through
+# the default feature extraction pipeline
+```
+
+## PHIL'S INPUT NEEDED HERE (A sample example for using funs and models)
+
 Please read the function documentation `?accelerometer_features` for more info.
 
 ## Gyroscope
```

Comment:
This is an exact copy paste + replace of the previous section on accelerometer data. Please consolidate these into a single section or write a short sentence or two stating that the functionality is the same and that the functional/data analogues are gyroscope_features/gyroscope_data.

Reply:
Yes it is. I thought if somebody skipped accelerometer and went to gyroscope directly, they would still get the whole vignette. 

I thought of consolidating, but we do have two different feature extraction functions for accelerometer and gyroscope, which we may make individual changes to down the line. But I worry that if I consolidate the vignette, people will point out why not do so with the functions too? I can summarize the gyroscope section by pointing them to the accelerometer section for further reference, but again I have the same worry. 

What is your call after this?

Reply:
The alternative is that they will worry "what are the differences?" if we have two distinct, but otherwise identical sections for each. There are differences between the two feature functions, but only in how column names of the transformed sensor data are handled, not the data itself.

Redundancy is fine and sometimes even good in documentation, but the redundancy here only conveys "that the functionality is the same and that the functional/data analogues are gyroscope_features/gyroscope_data" while obfuscating that same message with many more words. I recommend keeping the gyroscope section, but writing a short sentence or two stating that the functionality is the same and that the functional/data analogues are gyroscope_features/gyroscope_data.

Reply:
> The alternative is that they will worry "what are the differences?" if we have two distinct, but otherwise identical sections for each. There are differences between the two feature functions, but only in how column names of the transformed sensor data are handled, not the data itself.

Good point Phil, and point made.
> 
> Redundancy is fine and sometimes even good in documentation, but the redundancy here only conveys "that the functionality is the same and that the functional/data analogues are gyroscope_features/gyroscope_data" while obfuscating that same message with many more words. I recommend keeping the gyroscope section, but writing a short sentence or two stating that the functionality is the same and that the functional/data analogues are gyroscope_features/gyroscope_data.

I will make the gyroscope section concise, we can always add things later on if we make changes